Tmedia 731 tags block

JackHowa opened the pull request:
## Description

Update the tags block to v2 architecture

## Jira Ticket

- [TMEDIA-731](https://arcpublishing.atlassian.net/browse/TMEDIA-731) sub-task of [TMEDIA-713](https://arcpublishing.atlassian.net/jira/software/c/projects/TMEDIA/boards/875?modal=detail&selectedIssue=TMEDIA-713&assignee=5e387d6a1b1d910e5dfd7a9b)

## Acceptance Criteria

Tags Bar block is migrated to the new architecture.

Includes Custom fields of:

‘Lazy Load block?’ checkbox with help text

## Test Steps

1. Checkout this branch `git checkout TMEDIA-713-content-item-blocks`
2. Run fusion repo with linked blocks `npx fusion start -f -l @wpmedia/article-tag-block`
3. Go to http://localhost/pf/content-item-blocks/?_website=the-gazette. Compare with https://corecomponents-the-gazette-prod.cdn.arcpublishing.com/content-item-blocks/?_website=the-gazette and make sure that the gap matches the use of spacing before. 

## Effect Of Changes

### Before

- Matches v1 design

### After

- Pill font size matches the style designated in figma 

## Dependencies or Side Effects

- Discrepancies in font size are listed in discrepancies https://arcpublishing.atlassian.net/wiki/spaces/TI/pages/3138945094/Blocks+Design+Discrepancies
- feature pack https://github.com/WPMedia/arc-themes-feature-pack/pull/261
- manifest https://github.com/WPMedia/arc-themes-manifests/pull/61

## Author Checklist

_The author of the PR should fill out the following sections to ensure this PR is ready for review._

- [x] Confirmed all the test steps a reviewer will follow above are working.
- [x] Confirmed there are no linter errors. Please run `npm run lint` to check for errors. Often, `npm run lint:fix` will fix those errors and warnings.
- [x] Ran this code locally and checked that there are not any unintended side effects. For example, that a CSS selector is scoped only to a particular block.
- [x] Confirmed this PR has reasonable code coverage. You can run `npm run test:coverage` to see your progress.
  - [ ] Confirmed this PR has unit test files
  - [ ] Ran `npm run test`, made sure all tests are passing
  - [ ] If the amount of work to write unit tests for this change are excessive,
        please explain why (so that we can fix it whenever it gets refactored).
- [x] Confirmed relevant documentation has been updated/added.

## Reviewer Checklist

_The reviewer of the PR should copy-paste this template into the review comments on review._

- [ ] Linting code actions have passed.
- [ ] Ran the code locally based on the test instructions.
  - [ ] I don’t think this is needed to be tested locally. For example, a padding style change (storybook?) or a logic change (write a test).
- [ ] I am a member of the engine theme team so that I can approve and merge this. If you're not on the team, you won't have access to approve and merge this pr.
- [ ] Looked to see that the new or changed code has code coverage, specifically. We want the global code coverage to keep on going up with targeted testing.

# Question

- I should be able to override the flex-direction in the theme token file right? I feel like I'm missing something here.  -> So those default inline styles can't be overriden by the theme file. Instead, styles applied to those divs should use the inline styles.
<img width="1202" alt="Screen Shot 2022-05-13 at 16 55 12" src="https://user-images.githubusercontent.com/5950956/168395516-9354d24f-7d55-4a02-893c-f519117fd997.png">
- We're using line-height to establish height on the pill. If there's no text then there's no height. This is a change. Is this an ok change? Would we expect a different behavior if there's no text, like not rendering anything? https://www.chromatic.com/test?appId=5f91e4721ccaba0022a10782&id=6287b18eab872a003a89907a
- I removed the top and bottom spacing for the block itself. I also removed the border. Both of these should be handled in the future by a layout, right? https://www.chromatic.com/test?appId=5f91e4721ccaba0022a10782&id=6287b18eab872a003a899072

JackHowa committed:
feat: Declaring styles for tag override of flex direction for tag block

JackHowa committed:
style: Remove unnecessary style declaration for the desktop breakpoint

JackHowa commented:
> and for your question regarding the flex-direction we can use the prop direction="horizontal" within the component to get the tags to render horizontal as per current layout

Gotcha that makes sense to do in this case

nschubach commented on blocks/article-tag-block/package.json:
```diff
@@ -23,10 +24,7 @@
   },
   "gitHead": "83fb8d6685958d85ddbfcfbf01d9a5c864cd6c95"
-}
+}
```
```suggestion
}

```

JackHowa committed:
Merge branch 'arc-themes-release-version-2.0.1' into TMEDIA-713-content-item-blocks

JackHowa committed:
chore: Update deps and entry for v2 architecture

JackHowa committed:
test: Remove dependencies on engine theme sdk and other blocks

JackHowa committed:
feat: Use gap and theme styles along with new story with odd lengths text

JackHowa committed:
fix: Remove align center as not necessary

Commit:
Merge branch 'arc-themes-release-version-2.0.1' into TMEDIA-713-content-item-blocks

JackHowa committed:
style: Add back removed documentation for alias comments

JackHowa committed:
test: Update test for new block class name. Use new logic for pill component

Commit:
Merge branch 'arc-themes-release-version-2.0.1' into TMEDIA-713-content-item-blocks

Commit:
Update blocks/article-tag-block/package.json

Co-authored-by: nschubach <nschubach@gmail.com>

JackHowa committed:
docs: Use author per checklist

Commit:
Merge branch 'arc-themes-release-version-2.0.1' into TMEDIA-713-content-item-blocks

matthewroach commented on blocks/article-tag-block/_index.scss:
```diff
@@ -0,0 +1,10 @@
+@use "@wpmedia/arc-themes-components/scss";
+
+.b-article-tag {
+	&__item {
```
We don't need this class as it is applied to the `Pill` component which can be styled via the component overrides

JackHowa replied:
updated and removed

matthewroach commented on blocks/article-tag-block/features/tag/default.jsx:
```diff
@@ -1,39 +1,31 @@
-						href={href}
-						primaryColor={primaryColor || defaultBackgroundColor}
-					>
+					<Pill key={tag.text} href={href} className={`${BLOCK_CLASS_NAME}__item`}>
```
No need for className to be applied here as the user can control the styling via the component `Pill` overrides
```suggestion
					<Pill key={tag.text} href={href}>
```

JackHowa replied:
Removed -- thanks

matthewroach commented on blocks/article-tag-block/features/tag/default.jsx:
```diff
@@ -1,39 +1,31 @@
+				const { slug = "" } = tag;
+
+				// pill won't render an a tag if falsy href like ""
+				// instead renders a span
```
This is making an assumption of the underlying component, while it may use a `span` now that could change

```suggestion
```

matthewroach commented on blocks/article-tag-block/features/tag/default.test.jsx:
```diff
@@ -103,11 +98,14 @@ describe("the article tag block", () => {
 			const wrapper = mount(<ArticleTags />);
-			expect(wrapper.children().find("a").at(0).props().href).toBe("#");
-			expect(wrapper.children().find("a").at(1).props().href).toBe("#");
+			expect(wrapper.find("span").length).toBe(2);
```
This is testing the underlying component - we want to test the implementation not the element being rendered

JackHowa replied:
updated

matthewroach reviewed (dismissed):
A few comments added. 

For the Chromatic tests there are some differences between the new block and the current one, most notably the border above and below the block. Is this correct for the new version? Iff , what will be responsible for adding those borders back in?

JackHowa commented:
> For the Chromatic tests there are some differences between the new block and the current one, most notably the border above and below the block. Is this correct for the new version? Iff , what will be responsible for adding those borders back in?

I thought we would not want the block to be responsible for spacing between blocks, so I removed the border. 

I added a question in the pr review: 

> https://www.chromatic.com/test?appId=5f91e4721ccaba0022a10782&id=6287b18eab872a003a89907a - I removed the top and bottom spacing for the block itself. I also removed the border. Both of these should be handled in the future by a layout, right? https://www.chromatic.com/test?appId=5f91e4721ccaba0022a10782&id=6287b18eab872a003a899072

Commit:
Update blocks/article-tag-block/features/tag/default.jsx

Co-authored-by: Matthew Roach <matthewroach@users.noreply.github.com>

JackHowa committed:
docs: Remove comment as the interior logic of pill could change

Via https://github.com/WPMedia/arc-themes-blocks/pull/1347#discussion_r884621121

JackHowa committed:
refactor: Remove item as the component pill will be able to override

Via https://github.com/WPMedia/arc-themes-blocks/pull/1347#discussion_r884619568

JackHowa committed:
docs: Add back engine theme sdk as it's still being used

JackHowa committed:
docs: Reorder alphabetical

JackHowa committed:
test: Update tests to be less based on the implementation of components

Via https://github.com/WPMedia/arc-themes-blocks/pull/1347/files#r884622203

Commit:
Merge branch 'arc-themes-release-version-2.0.1' into TMEDIA-713-content-item-blocks

JackHowa commented:
Based on direction of the ticket, the spacing difference and border would be handled in a new ticket based on design discretion.

JackHowa committed:
docs: Update to use pills component

JackHowa committed:
Merge branch 'TMEDIA-713-content-item-blocks' of github.com:WPMedia/fusion-news-theme-blocks into TMEDIA-713-content-item-blocks

brentmiller1973 reviewed (approved):
Based on discussions yesterday that border and spacing would be attended to at a later time, then this looks good to me.

matthewroach commented on blocks/article-tag-block/features/tag/default.test.jsx:
```diff
@@ -53,26 +48,26 @@ describe("the article tag block", () => {
 		});
 
 		it("should render a parent container for the tags", () => {
 			const { default: ArticleTags } = require("./default");
```
Are we able to convert these tests to use the `import` syntax instead of the `require` syntax?

JackHowa replied:
yep updated!

JackHowa replied:
yep updated! thanks for the feedback @matthewroach

JackHowa committed:
test: Update tests to use import syntax

Via https://github.com/WPMedia/arc-themes-blocks/pull/1347#discussion_r887009311

JackHowa committed:
test: Update to mock return value of fallback
